packages/clay-core/docs/treeview.mdx

```diff
@@ -0,0 +1,378 @@
+	<TreeView.ItemStack>
+		<Checkbox />
+		<Icon symbol="folder" />
+		Drive
+	</TreeView.ItemStack>
+	<TreeView.Group>
+		{...}
+	</TreeView.Group>
+</TreeView.Item>
+```
+
+### Actions
+
+// TODO
+
+## Expander
+
+The expansion of an Item in the TreeView is internally controlled and it is possible to control the state and set an initial value with the keys of the items that are initially expanded.
+
+The TreeView exposes the `expandedKeys` properties and the `onExpandedChange` event to control the state of expanded Items, the `expandedKeys` must be a collection of values using the `Set()` object.
+
+```jsx
+function Example() {
+	const [expandedKeys, setExpandedKeys] = useState(
+		new Set(['1', '2', '3'])
+	);
+
+	return (
+		<TreeView
+			expandedKeys={expandedKeys}
+			onExpandedChange={setExpandedKeys}
+		>
+			{...}
+		</TreeView>
+	);
+}
+```
+
+### Custom expander
+
+Customizing the expander is possible using the `expanderIcons` property, changing icons has an effect on the entire tree, it is not possible to change exclusively for a single Item.
+
+```jsx
+<TreeView
+	expanderIcons={{
+		close: <Icon symbol="hr" />,
+		open: <Icon symbol="plus" />,
+	}}
+>
+	{...}
+</TreeView>
+```
+
+## Multiple Selection
+
+The multi-selection in TreeView is a composition with the `<Checkbox />` component that must be added explicitly in the Item rendering.
+
+```jsx
+<TreeView.Item>
+	<Checkbox />
+	Drive
+</TreeView.Item>
+
+<TreeView.Item>
+	<TreeView.ItemStack>
+		<Checkbox />
+		Drive
+	</TreeView.ItemStack>
+	<TreeView.Group>
+		{...}
+	</TreeView.Group>
+</TreeView.Item>
+```
+
+It is not necessary to add the `onChange` event or the `checked` property, the TreeView adds the methods to the component under the covers, this allows you to just add the Checkbox in any order you want.
+
+Multi-selection is also controlled in the same way as Expander, it uses the `key` of the item to select the Item, the `selectedKeys` property and the `onSelectionChange` event are used for this purpose.
+
+```jsx
+function Example() {
+	const [selectedKeys, setSelectionChange] = useState(
+		new Set(['1', '2', '3'])
+	);
+
+	return (
+		<TreeView
+			onSelectionChange={setSelectionChange}
+			selectedKeys={selectedKeys}
+		>
+			{...}
+		</TreeView>
+	);
+}
+```
+
+### Recursive selection
+
+By default when selecting an Item with nested items, the items are recursively selected.
+
+There are some limitations, for static content the recursive selection only works if the item is expanded, ie visible. For dynamic content, it works independently.
+
+<div class="clay-site-alert alert alert-warning">
+	<strong class="lead">Warning</strong>
+	Recursive selection will not select items from an asynchronous Item.
+</div>
+
+<MultipleSelection />
+
+## Drag and Drop
+
+The Drag and Drop implementation handles items internally and works only for TreeView with dynamic content because it depends on the `items` property and must be enabled using the `dragAndDrop` property.
```

Comment:
```suggestion
The drag and drop implementation handles items internally and works only for TreeView with dynamic content because it depends on the `items` property and must be enabled using the `dragAndDrop` property.
```